devtools/x-lint/src/project.rs

```diff
@@ -12,31 +13,35 @@ pub trait ProjectLinter: Linter {
     /// Executes the lint against the given project context.
     fn run<'l>(
         &self,
-        ctx: &ProjectContext<'l>,
+        ctx: ProjectContext<'l>,
```

Comment:
I'm not sure I understand this change, now that you're passing this by value and changed the fns below to take by `self` instead of `&self` wouldn't you only be able to call a single function and then consume the thing....wait

It impls Copy so this is actually ok but took me a second to realize that :) Is there a particular reason you decided to use Copy types for these context structs?

Reply:
clippy warned about it :(

Reply:
I agree that consistency is more important so I'll probably revert it and disable the clippy lint

Reply:
Done.